public/app/features/dashboard/components/DashExportModal/DashboardExporter.test.ts

```diff
@@ -94,6 +94,41 @@ it('handles a default datasource in a template variable', async () => {
   expect(exported.templating.list[0].datasource.uid).toBe('${DS_GFDB}');
 });
 
+it('do not expose datasource name and id in a in a template variable of type datasource', async () => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const dashboard: any = {
+    templating: {
+      list: [
+        {
+          current: {
+            selected: false,
+            text: 'my-prometheus-datasource',
+            value: 'my-prometheus-datasource-uid',
+          },
+          hide: 0,
+          includeAll: false,
+          multi: false,
+          name: 'query1',
+          options: [],
+          query: 'prometheus',
+          queryValue: '',
+          refresh: 1,
+          regex: '',
+          skipUrlSync: false,
+          type: 'datasource',
+        },
+      ],
+    },
+  };
+  const dashboardModel = new DashboardModel(dashboard, undefined, {
+    getVariablesFromState: () => dashboard.templating.list,
+  });
+  const exporter = new DashboardExporter();
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Same here. You should remove this. There's a bad behavior in `makeExportable` function , because it returns either an `ExternalDashboard` or `{error: unknown}`.
We should work on improving that `makeExportable` signature. In the meantime, I would cast it to `ExternalDashboard`. 
On the other hand, we don't have `templating` as an attribute of `ExternalDashboard`.
I remember I modified that `ExternalDashboard` interface because absolutely **anything** was typed and I was working on a library panel issue when exporting the json. I decided to add just the values I needed as optional, because I didn't know what to expect.
Therefore, I would add the templating attribute as a new one

```
export interface ExternalDashboard {
  __inputs?: Input[];
  __elements?: Record<string, LibraryElementExport>;
  __requires?: Array<Requires[string]>;
  panels: Array<PanelModel | PanelWithExportableLibraryPanel>;
  templating?: {
    list?: VariableModel[];
  };
}
```